### PR TITLE
feat(rules): add rule creation form and deletion action

### LINKED ISSUE #377: Automated Rules creation/deletion

### PR DESCRIPTION
**Pull request by andrewazores**
Fixes #377

For CORS/devserver testing you may need to use https://github.com/cryostatio/cryostat/pull/838 on the backend.

### CODE REVIEW

File: src/app/Rules/CreateRule.tsx [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+                onChange={handleTemplateChange}
+                aria-label="Event Template Input"
+              >
+                { templatesOptionGroups }
+              </FormSelect>
+            </FormGroup>
+            <FormGroup
+              label="Maximum Size"
+              fieldId="maxSize"
+              helperText="The maximum size of recording data retained in the applictarget application's recording buffer."
```
Comment (jan-law) [inline src/app/Rules/CreateRule.tsx:261]: typo

File: src/app/Rules/CreateRule.tsx [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+  }, []);
+
+  const breadcrumbs: BreadcrumbTrail[] = [
+    {
+      title: 'Automated Rules',
+      path: '/rules',
+    }
+  ];
+
+  const templatesOptionGroups = React.useMemo(() => {
```
Comment (jan-law) [inline src/app/Rules/CreateRule.tsx:170]: Is this the same component used in the custom recording form? If so, could we extract and reuse it somehow?

File: src/app/Rules/CreateRule.tsx [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+
+  return (
+    <BreadcrumbPage pageTitle='Create' breadcrumbs={breadcrumbs} >
+      <Card>
+        <CardBody>
+          <Form isHorizontal>
+            <Text component={TextVariants.small}>
+              Automated Rules are configurations that instruct Cryostat to create JDK Flight Recordings on matching target JVM applications
+              with specified parameters for which Event Template to use, how much data should be kept in the application recording buffer, and
+              how frequently Cryostat should copy the application recording buffer into Cryostat's own archived storage.
```
Comment (jan-law) [inline src/app/Rules/CreateRule.tsx:195]: Could you split this sentence into two sentences for clarity? Maybe something like "Automated Rules are configurations that instruct Cryostat to create JDK Flight Recordings on matching target JVM applications. Each Automated Rule specifies parameters for which Event Template to use, how much data should be kept in the application recording buffer, and how frequently Cryostat should copy the application recording buffer into Cryostat's own archived storage."

**Review by jan-law — COMMENTED** [review]
This is working well. I'm also assuming when you implement the match expression validation, there'll be some helper text to indicate the correct syntax for that field.

**Review by hareetd — COMMENTED** [review]
Works well.

**Comment by andrewazores** [comment]
> This is working well. I'm also assuming when you implement the match expression validation, there'll be some helper text to indicate the correct syntax for that field.

Yea, some helper text and some kind of tie-in to #376 I think. I'll have to experiment with that feature to see what's feasible and how I can present it.

**Review by jan-law — APPROVED** [review]

**Review by hareetd — APPROVED** [review]

**Pull request merged**